SnowplowTracker/SnowplowTracker/Requests/UnityWebRequestAwaiter.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace SnowplowTracker.Requests
+{
+	public class UnityWebRequestAwaiter : INotifyCompletion

```

Comment:
Is this class necessary? If so, what is used for?

Reply:
Yes, this class is necessary to make `UnityWebRequest` method awaitable. It could be that we could make it private, and only have the `ExtensionMethods` as public, but unsure. 

In my PR description, I also mention that we can bypass this if we add [UniTask](https://github.com/Cysharp/UniTask) as a depedency (which also makes handling tasks in Unity much better)

Reply:
Sorry, I forgot about that since reading the PR description. I think this is sufficient for now, we can add the UniTask dependency later when we see it necessary.